setup.py

```diff
@@ -375,7 +375,9 @@ def _quote_build_define(argument):
     elif LINUX_AARCH64 == boringssl_asm_platform:
         asm_key = 'crypto_linux_aarch64'
     elif "mac" in boringssl_asm_platform and "x86_64" in boringssl_asm_platform:
-        asm_key = 'crypto_mac_x86_64'
+        asm_key = 'crypto_apple_x86_64'
+    elif "mac" in boringssl_asm_platform and "arm64" in boringssl_asm_platform:
```

Comment:
Is the `"mac" in boringssl_asm_platform and "arm64" in boringssl_asm_platform` combination actually used anywhere at this point or is it just a proactive change?
Also is "arm64" the right value to use for mac? (linux uses "aarch64")

CC @HannahShiSFB @gnossen

Reply:
Needed for M1 mac.
This should fix https://github.com/grpc/grpc/issues/28433 as well

Reply:
AFAIK we don't publish python wheels for M1 Macs currently, so it sounds like this is being proactive.
Anyway, the entire section that select the asm_key can be simplified as a followup (since we the asm files are `#ifdef` conditioned to only kick in on the right platform, it's no longer important so select only the asm files that exactly match the target platforms, we can simply determine whether we want the asm optimizations or not, set OPENSSL_NO_ASM accordingly and include all gcc-style asm files in the build).

Since likely want the followup simplification, it's fine to leave this change as is.